ledger/src/snapshot_utils.rs

```diff
@@ -349,6 +418,93 @@ where
+                context_bankrc_to_stream::<SerdeContextV1_1_0, _>(
+                    stream.by_ref(),
+                    &bank.rc,
+                    snapshot_storages,
+                )?;
+                Ok(())
+            },
+        )),
+        SnapshotVersion::V1_1_1 => Ok(Box::new(
+            move |stream: &mut BufWriter<File>| -> Result<()> {
+                serialize_into(stream.by_ref(), bank)?;
+                context_bankrc_to_stream::<SerdeContextV1_1_1, _>(
+                    stream.by_ref(),
+                    &bank.rc,
+                    snapshot_storages,
+                )?;
+                Ok(())
+            },
+        )),
+        //If additional snapshot versions were defined but not implemented...
+        //_ => Err(get_io_error(&format!("unsupported snapshot version: {}", snapshot_version.to_string()))),
+    }
+}
+
+#[allow(clippy::type_complexity)]
+fn create_versioned_bank_snapshot_deserializer<'a, P: AsRef<Path>>(
+    snapshot_version: &str,
+    account_paths: &'a [PathBuf],
+    frozen_account_pubkeys: &'a [Pubkey],
+    append_vecs_path: &'a P,
+    genesis_config: &GenesisConfig,
+) -> Result<Box<dyn Fn(&mut BufReader<File>) -> Result<Bank> + 'a>> {
+    let operating_mode = genesis_config.operating_mode;
+    match SnapshotVersion::maybe_from_string(snapshot_version) {
+        Some(SnapshotVersion::V1_1_0) => Ok(Box::new(move |stream: &mut BufReader<File>| {
+            let mut bank: Bank = bincode::config()
+                .limit(MAX_SNAPSHOT_DATA_FILE_SIZE)
+                .deserialize_from(stream.by_ref())?;
+            bank.operating_mode = Some(operating_mode);
+            info!("Rebuilding accounts...");
+            let rc = context_bankrc_from_stream::<SerdeContextV1_1_0, _, _>(
+                account_paths,
+                bank.slot(),
+                &bank.ancestors,
+                frozen_account_pubkeys,
+                stream.by_ref(),
+                &append_vecs_path,
+            )?;
+            bank.rc = rc;
+            bank.finish_init();
+            Ok(bank)
+        })),
+        Some(SnapshotVersion::V1_1_1) => Ok(Box::new(move |stream: &mut BufReader<File>| {
+            let mut bank: Bank = bincode::config()
+                .limit(MAX_SNAPSHOT_DATA_FILE_SIZE)
+                .deserialize_from(stream.by_ref())?;
+            bank.operating_mode = Some(operating_mode);
+            info!("Rebuilding accounts...");
+            let rc = context_bankrc_from_stream::<SerdeContextV1_1_1, _, _>(
+                account_paths,
+                bank.slot(),
+                &bank.ancestors,
+                frozen_account_pubkeys,
+                stream.by_ref(),
+                &append_vecs_path,
+            )?;
+            bank.rc = rc;
+            bank.finish_init();
+            Ok(bank)
+        })),
+        _ => Err(get_io_error(&format!(
+            "unsupported snapshot version: {}",
+            snapshot_version
+        ))),
+    }
```

Comment:
Ideally, `frozen_account_pubkey` and `generate_index` code could be moved to here?